_episodes/41-code-review.md

```diff
@@ -208,286 +148,468 @@ and update your proposed changes within a self-contained bundle.
+   and the compare branch contains your changes.
+5. Click `Create pull request` to open the request
+   ![Creating a new pull request.](../fig/github-create-pull-request.png){: .image-with-shadow width="900px"}
+6. Add a comment describing the nature of the changes,
+   and then submit the pull request by clicking `Create pull request`.
+![Submitting a pull request.](../fig/github-submit-pullrequest.png){: .image-with-shadow width="900px"}
+7. At this point, the code review process is initiated.
+
+We will now discuss what to look for in a code review,
+before practising this on this fictional change.
+
+## Things to look for in a code review
+
+Reviewing code effectively takes practice.
+However, here is some guidance on the kinds of things you should
+be looking for when reviewing a piece of code.
+
+Start by understanding what the code _should_ do, by reading the specification/user requirements,
+the pull request description and talking to the developer.
+In this case, understand what [SR1.1.1](../31-software-requirements/index.html#solution-requirements) means.
+
+Once you're happy, start reading the code (skip the test code for now). You're
+going to be assessing the code in 4 key areas:
+
+* Is the code readable
+* Is the code a minimal change
+* Is the structure of the code clear
+* Is there appropriate and up-to-date documentation
+
+### Is the code readable
+
+Think about do the names of the variables, do they [follow guidelines for good
+names?](../15-coding-conventions/index.html#l#naming-conventions)
+
+Do you understand what conditions in each if statements are for?
+
+Do the function names match the behavior of the function.
+
+### Is the code a minimal change
+
+Does the code reimplement anything that already exists, either
+elsewhere in the codebase or in a library you know about?
+
+Does the code implement something that isn't on the ticket?
+
+### Is the structure of the code clear
+
+Do functions do just one thing? Have appropriate design
+patterns been used (e.g. separating out the model logic from
+any view considerations)? Is the code consistent with the
+structure of the rest of the code?
+
+### Is there appropriate and up-to-date documentation
+
+If functionality has changed, has corresponding documentation been
+updated? If new functions have been added, do they have appropriate
+levels of documentation? Does the documentation make sense?
+
+Are there clear and useful comments that explain complex designs
+and focus on the "why/because" rather than the "what/how"?
+
+## Effective review comments
+
+Make sure your review comments are specific and actionable.
+
+Try to be as specific as you can, rather than "this code is unclear"
+prefer, "I don't understand what values this variable can hold".
+
+Make it clear in the comment if you want something to change as part
+of this pull request.
+Ideally provide a concrete suggestion (e.g. better variable name).
+
+## Adding a review comment
```

Comment:
Feels like we need "what steps are of the code review?" to highlight the process earlier in the episode. So, e.g. reviewing the code and what to look for, commenting on code, buffering all comments and finishing/sending off our review, responding to review comments. Or something like that.

Reply:
I've added a diagram in the code review in action section which will hopefully address this comment - let me know if you think we need more.